core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -765,12 +765,38 @@ class WorkspaceService(protected val userInfo: UserInfo,
               )
             }
           }
-          workspaces.map { workspace =>
+
+        // Internal folks may create MCWorkspaces in local WorkspaceManager instances, and those will not
+        // be reachable when running against the dev environment.
+        val reachableWorkspaces = workspaces.filter { workspace =>
+          workspace.workspaceType match {
+            case WorkspaceType.McWorkspace =>
+              try {
+                workspaceManagerDAO.getWorkspace(workspace.workspaceIdAsUUID, userInfo.accessToken)
+                true
+              } catch {
+                case e: ApiException =>
+                  logger.warn(s"MC Workspace ${workspace.name} (${workspace.workspaceIdAsUUID}) does not exist in the current WSM instance. ")
+                  false
+              }
+            case _ => true
+          }
+        }
+
+          reachableWorkspaces.map { workspace =>
+            val cloudPlatform = workspace.workspaceType match {
```

Comment:
It looks like we're making a second call to `getWorkspace` here, is that redundant given the one above on L775?

Reply:
It's not redundant unless we store the first result from `getWorkspace` or do some sort of combination filtering/mapping. I couldn't figure out how to do the exception handling with a `collect`, but I thought Scala experts might have some ideas.

On the other hand, I would expect that the `getWorkspace` call is O(n), and thus calling it twice shouldn't introduce much inefficiency. @aherbst-broad am I incorrect about the cost of calling `getWorkspace`?